module/data/abstract.mjs

```diff
@@ -0,0 +1,79 @@
+/**
+ * Data Model variant with some extra methods to support template mix-ins.
+ *
+ * **Note**: This uses some advanced Javascript techniques that are not necessary for most data models.
+ * Please refer to the [advancement data models]{@link BaseAdvancement} for an example of a more typical usage.
+ *
+ */
+export default class SystemDataModel extends foundry.abstract.DataModel {
+
+  /**
+   * Base templates used for construction.
+   * @type {*[]}
+   * @protected
```

Comment:
We should consider `@internal` here. I don't think subclasses should really be messing with this property, it's exclusively part of the mixin mechanism.

Reply:
`@internal` isn't a JSDoc property. We currently don't have TSDoc definitions, so that throws a linting error.